### PR TITLE
Add change to timer

### PR DESCRIPTION
**Pull request by HassanHashemi**
null

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/orleans/pull/7396)

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/orleans?pullRequest=7396) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/orleans?pullRequest=7396) before we can accept your contribution.<br/><br/>:x: HassanHashemi [sign now](https://cla.dotnetfoundation.org/dotnet/orleans?pullRequest=7396)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/orleans?pullRequest=7396) it.</sub>

File: src/Orleans.Core.Abstractions/Runtime/IGrainTimer.cs [hunk 1]
```diff
@@ -3,12 +3,14 @@
...
 namespace Orleans.Runtime
 {
-    internal interface IGrainTimer : IDisposable
+    public interface IGrainTimer : IDisposable
     {
         void Start();
 
         void Stop();
 
         Task GetCurrentlyExecutingTickTask();
```
Comment (ReubenBond) [inline src/Orleans.Core.Abstractions/Runtime/IGrainTimer.cs:12]: I think the above three methods are less relevant to callers and they shouldn't use them. Therefore, I wonder if the existing `IGrainTimer` should be renamed to `IInternalGrainTimer` and the `Change` method added to a new `IGrainTimer`

Comment (HassanHashemi) [reply src/Orleans.Core.Abstractions/Runtime/IGrainTimer.cs:12]: Yes, I'll change it ASAP.

**Comment by ReubenBond** [comment]
We've had others request this feature, too. I think it's good to have.

**Comment by ReubenBond** [comment]
I'll close this due to idleness for now. Please feel free to re-open, addressing the above comments. Also note #7605

**Pull request closed without merging**